### PR TITLE
Move some functions from rancheros-operator

### PR DESCRIPTION
**Pull request by ldevulder**
As these functions could be useful in other tests as well.

Related to https://github.com/rancher-sandbox/rancheros-operator/pull/21.

### CODE REVIEW

File: kubectl/kubectl.go [hunk 1]
```diff
@@ -87,21 +87,85 @@ func (k *Kubectl) GetPodNames(namespace string, selector string) ([]string, erro
...
+	pods, err := k.GetPodNames(namespace, labelName)
+	if err != nil {
+		return false, err
+	}
+	for _, p := range pods {
+		e, _ := k.PodStatus(namespace, p)
+		if e == nil || e.ContainerStatuses == nil || len(e.ContainerStatuses) == 0 {
+			return false, nil
+		}
+		if !e.ContainerStatuses[0].Ready {
```
Comment (mudler) [inline kubectl/kubectl.go:108]: This assumes there is one container in the pod, maybe we can extend that for each container in the pod?

**Review by mudler — APPROVED** [review]
Besides small nits it looks good!

File: kubectl/kubectl.go [hunk 2]
```diff
@@ -87,21 +87,85 @@ func (k *Kubectl) GetPodNames(namespace string, selector string) ([]string, erro
...
 
+// WaitForNamespaceWithPod blocks until pods matching the selector are available in the specified namespace. It fails after the timeout.
+func (k *Kubectl) WaitForNamespaceWithPod(namespace string, labelName string) error {
+	return wait.PollImmediate(k.pollInterval, k.PollTimeout, func() (bool, error) {
+		return k.NamespaceWithPodExists(namespace, labelName)
+	})
+}
+
+// NamespaceWithPodExists returns true if pods by that label are present in the given namespace
+func (k *Kubectl) NamespaceWithPodExists(namespace string, labelName string) (bool, error) {
```
[the remark this answers is not in the document]
Comment (ldevulder) [reply kubectl/kubectl.go:98]: Thanks, was not sure about the function name, YBH I iterate 3 ou 4 times on this one :D

File: kubectl/kubectl.go (hunk 1 above)
Comment (ldevulder) [reply kubectl/kubectl.go]: Sure, the last commit should do the job.

**Comment by ldevulder** [comment]
@mudler do you think that this can be merged?

**Review by mudler — APPROVED** [review]

**Pull request merged**